### DISCUSSION TITLE
USB chip

### DISCUSSION THREAD

**Discussion by KrystianD** [ludwich66/RC3563#2] [Ideas]
Hey, great to see the wiki for this device!

I got my RC device without USB support but I managed to resolder the proper chip from the NodeMCU board.

The chip that is compatible is **CP2102** - you may want to include this information in your wiki as well, in case others also receive thier unit lacking this feature..

![image](https://user-images.githubusercontent.com/1083568/192846961-ee9c29d0-bfb4-42a5-b8b4-1333fe02a798.png)
![image](https://user-images.githubusercontent.com/1083568/192846979-3cb8b654-5059-440f-91f4-ec28f4bbf15b.png)
![image](https://user-images.githubusercontent.com/1083568/192847447-4852d42d-fb95-4657-9267-92956d8723f0.png)

**Comment by ludwich66** [comment]
Hi Krystian,

Great, do you have  more pictures? (Better align ;-)

You Mod is to add a CP2102 on the right place?
without other parts or software mods?

I was just curious and opened my device.
I have a "newer" board revision with a CH340 chip in a different layout.

My software version is quite new 2022-09-28 V10.2
Board marker 20220319

What kind do you have?
software revision:
board number:
Did you see differences in menue? (see my wiki)

I will compare the circuit boards, can you send me pictures of your device in a suitable layout?
I will add it to the Wikipage.

Do you have any other requests or suggestions for the wiki?

greetings ludwich

**Comment by KrystianD** [comment]
> Great, do you have  more pictures? (Better align ;-)

Please see all pictures I have: https://send.vis.ee/download/d094679ee7bb43e0/#W68bZi7tBNsVSooR440gnQ, probably not of much use, I can take a better one later.

> You Mod is to add a CP2102 on the right place?

Correct, I wanted to use it in my automated system for testing 18650 cells, so I wanted PC communication, but it was not working so I started investigating. The issue was lack of the USB<->UART chip.

> without other parts or software mods?

Nothing besides soldering the chip was required, fortunately :)

> What kind do you have?

Regarding version - "About" says:
`2021-08-09 v8.0`

Board seems to be marked: `20210101`

> Did you see differences in menue? (see my wiki)

Menu seems similar

**Comment by ludwich66** [comment]
Thanks,

first i added Picture. 
Your Mod follows.
Is it possible to describe me the Softwareprotocoll in human language?

in style:
…
read 10 bytes
filter  ranges …
scale
print

More tomorrow ;-)

ludwich



> Am 28.09.2022 um 20:34 schrieb Krystian Dużyński ***@***.***>:
> 
> 2021-08-09 v8.0

**Comment by KrystianD** [comment]
From https://gist.github.com/mgrela/e5e52e22e5d748645bf4d0ebe2c133dd

I find this protocol unnecessarily convoluted, but very briefly:

4 bits - resistance unit + overflow info - 5 = mΩ, 6 = mΩ (overflow), 9 = Ω, 10 = Ω (overflow)
4 bits - voltage overflow info, 4 = all ok, 8 = overflow
1 byte - selected resistance range - 1 = 20 mΩ, 2 = 200 mΩ, 3 = 2 Ω, 4 = 20 Ω, 5 = 200 Ω, 6 = AUTO
3 bytes - forming 3-byte integer with the resistance value
1 byte - voltage sign value, 1 = positive, 0 = negative
1 byte - selected voltage range, 1 = 20V, 2 = 100V, 3 = AUTO
3 bytes - forming 3-byte integer with the voltage value

something like this

**Comment by ludwich66** [comment]
Next Part filled  +++

Ludwich

> Am 28.09.2022 um 20:57 schrieb Krystian Dużyński ***@***.***>:
> 
> 4 bits - resistance unit + overflow info - 5 = mΩ, 6 = mΩ (overflow), 9 = Ω, 10 = Ω (overflow)
> 4 bits - voltage overflow info, 4 = all ok, 8 = overflow
> 1 byte - selected resistance range - 1 = 20 mΩ, 2 = 200 mΩ, 3 = 2 Ω, 4 = 20 Ω, 5 = 200 Ω, 6 = AUTO
> 3 bytes - forming 3-byte integer with the resistance value
> 1 byte - sign value, 1 = positive, 0 = negative
> 1 byte - selected voltage range, 1 = 20V, 2 = 100V, 3 = AUTO
> 3 bytes - forming 3-byte integer with the voltage value
> 
>

**Comment by ludwich66** [comment]
Krystian,

please look to Wiki, section Mods.

Is it OK for You?

Ludwich

**Reply by KrystianD** [reply]
it is perfect, thanks!

**Comment by ludwich66** [comment]
Thanks ;-)

Another Questions
Measurement rate ist described with 5/sec.

But its a continious Stream with aproximate 4 to 6 Conversitions per second?

Can you say something about the running time of the device?
How long does the lipo last?
You're probably plugged into the USB port…

Ludwich

**Reply by KrystianD** [reply]
Actually, in the end I use this device occasionally as a milliohm meter for calibration or testing wires, nickel plates etc, as I created my own [Modular Liion Servicer](https://krystiand.net/gallery/01-Solar-system-MLS.html) with AC impedance meter support built-in.

But its lipo is quite good, I think I charged it one or two times in last year (used very occasionally). I am not sure about the rate, but it is much faster than typical voltmeter so its good.

**Comment by ludwich66** [comment]
Oh a great project!

Brilliant what you can produce with small quantities thanks to the new printed circuit board production.

Did you use same Methode to meassure?

I wanted to measure some coils and wasn't sure if there would be problems with induction voltage at 1kHz.
But I'm new to the device and have little experience.
Do you see that as a walking hazard?

Ludwich

**Reply by KrystianD** [reply]
> Did you use same Methode to meassure?

I am not sure what method they use in this device, but I am getting same results :) basically, I put 1 kHz signal with audio op-amp and checking the voltage drop with high-pass filter.

> Do you see that as a walking hazard?

I don't think it would break the device, but it may introduce a bit of a phase delay to the current, which - depending on the algorithm in RC3563 - may false the results a bit.

**Comment by ludwich66** [comment]
Thank you for your assessment ;-)

CU

Ludwich

> Am 28.09.2022 um 22:24 schrieb Krystian Dużyński ***@***.***>:
> 
> Did you use same Methode to meassure?
> 
> I am not sure what method they use in this device, but I am getting same results :) basically, I put 1 kHz signal with audio op-amp and checking the voltage drop with high-pass filter.
> 
> Do you see that as a walking hazard?
> 
> I don't think it would break the device, but it may introduce a bit of a phase delay to the current, which - depending on the algorithm in RC3563 - may false the results a bit.
> 
>
